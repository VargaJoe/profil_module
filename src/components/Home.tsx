import * as React				            from 'react';
import { PathHelper }                       from '@sensenet/client-utils';
import { connect }                          from 'react-redux';
import { Actions }                          from '@sensenet/redux';
import { KnowledgeBaseArticleV2 }           from '../type/KnowledgeBaseArticle_v_2';
import { IODataParams }                     from '@sensenet/client-core';
import Loader                               from './Loader';

const DATA = require('../config.json');

export interface Props {
	getHomeContent: 		(path: string, options: IODataParams<KnowledgeBaseArticleV2>) => Promise<{
		action: any;
		value: any;
	}>;
}

export class HomeComponent extends React.Component<Props, any> {
    constructor(props: any) {
        super(props);
        this.state = {
            isDataFetched: false,
            articles: {},
        };
    }

    async componentDidMount  () {
        let path = PathHelper.joinPaths(DATA.home);
		// get the current user info
		let userGet = await this.props.getHomeContent(path, {
            query: 'TypeIs:KnowledgeBaseArticle_v_2',
		});
        
        this.setState({ 
            isDataFetched : true,
            articles: userGet.value.entities.entities
        });
    }
    
    render() {
        if ( !this.state.isDataFetched ) {
            return (<Loader/>);
		}
        
        let homePageItems = this.state.articles;
        const homePage = Object.keys(homePageItems).map( (key: any) => 
            (
                <div key={key}>
                    <h2>{homePageItems[key].DisplayName}</h2>
                    <div>{homePageItems[key].Description}</div>
                </div>
            )
        );

		return (
			<div className="about">
				<h1>Home Page</h1>
                {homePage}
			</div>
		);
	}
}

const mapStateToProps = (state: any, match: any) => {
    return {
    };
};

export default connect(
    mapStateToProps,
    (dispatch) => ({
        getHomeContent:    (path: string, options: any) => dispatch(Actions.requestContent( path, options )),
    })
)(HomeComponent as any);
 