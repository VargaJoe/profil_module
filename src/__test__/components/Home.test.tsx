import * as React 						  	from 'react';
import Home ,
		{ HomeComponent }					from '../../components/Home';
import { 
	configure, 
		shallow,
		mount, 
		ReactWrapper, 
		ShallowWrapper } 					from 'enzyme';
import * as createMockStore					from 'redux-mock-store';
import * as Adapter 					  	from 'enzyme-adapter-react-16';
import toJson 				      			from 'enzyme-to-json';
import { Provider } from 'react-redux';

configure( {adapter: new Adapter()} );

describe('<Home /> shallow rendering', () => {
	let store;
	let homeMount: ReactWrapper<any, any>;
	let homeComponentMount: ReactWrapper<any, any>;
	let homeShallow: ShallowWrapper<any, any>;

	const mockStore = (createMockStore as any)();
	beforeEach( () => {
		let getHomeContent: HomeComponent['props']['getHomeContent'] = async () => {
            return {
				action: {},
				result: {}
            };
		};
        store = mockStore({
			sensenet: {
				session: {
					loginState: 'Unauthenticated',
					user: {
						userName: 'Visitor'
					}
				}
            },
			user: {
				user: {
					Id: 12312,
					FullName: 		'adasdasdasd',
					JobTitle: 		'adasdasdasd',
					Email: 			'adasdasdasd',
					Languages: 		'adasdasdasd',
					Phone: 			'adasdasdasd',
					BirthDate: 		'adasdasdasd',
					Education: 		'adasdasdasd',
					Description: 	'adasdasdasd',
				}
			}
		});
		const user = {
			Id: 12312,
			FullName: 		'adasdasdasd',
			JobTitle: 		'adasdasdasd',
			Email: 			'adasdasdasd',
			Languages: 		'adasdasdasd',
			Phone: 			'adasdasdasd',
			BirthDate: 		'adasdasdasd',
			Education: 		'adasdasdasd',
			Description: 	'adasdasdasd',
		};
		homeMount = mount(
			<Provider store={store}>
				<Home/>
			</Provider>); 
		homeComponentMount = mount(
			<Provider store={store}>
				<Home 
					getHomeContent={getHomeContent}
				/>
			</Provider>); 
		homeShallow = shallow(
			<Provider store={store}>
				<Home 
					getHomeContent={getHomeContent}
				/>
			</Provider>); 
	});
		
	// test Snapshot 
	it('Match to snapshot', () => {
		expect(toJson(homeMount)).toMatchSnapshot();
		expect(toJson(homeComponentMount)).toMatchSnapshot();
	});

	// test Snapshot 
	it('Match to snapshot', () => {
		homeShallow.setState({
			isDataFetched: false
		});
		expect(toJson(homeShallow)).toMatchSnapshot();
	});

	it('Test componentDidMount', () => {
		const spy = jest.spyOn(homeMount.children().instance(), 'componentDidMount');
		const ddd = homeMount.children().instance().componentDidMount();
		expect(spy).toHaveBeenCalledTimes(1);
	});
		 
});
 