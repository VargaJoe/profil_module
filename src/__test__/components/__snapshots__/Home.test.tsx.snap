// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> shallow rendering Match to snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(HomeComponent)>
    <HomeComponent
      getHomeContent={[Function]}
    >
      <Loader>
        <div
          className="loader"
        >
          <div
            className="leftEye"
          />
          <div
            className="rightEye"
          />
          <div
            className="mouth"
          />
        </div>
      </Loader>
    </HomeComponent>
  </Connect(HomeComponent)>
</Provider>
`;

exports[`<Home /> shallow rendering Match to snapshot 2`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(HomeComponent)
    getHomeContent={[Function]}
  >
    <HomeComponent
      getHomeContent={[Function]}
    >
      <Loader>
        <div
          className="loader"
        >
          <div
            className="leftEye"
          />
          <div
            className="rightEye"
          />
          <div
            className="mouth"
          />
        </div>
      </Loader>
    </HomeComponent>
  </Connect(HomeComponent)>
</Provider>
`;

exports[`<Home /> shallow rendering Match to snapshot 3`] = `
<Connect(HomeComponent)
  getHomeContent={[Function]}
/>
`;
